src/dotnet-new3/CompleteCommand.cs

```diff
@@ -23,11 +24,11 @@ internal CompleteCommand() : base("complete", "tab completion")
         {
             this.AddArgument(PathArgument);
             this.AddOption(PositionOption);
-            this.Handler = CommandHandler.Create<ParseResult>(Run);
+            this.Handler = CommandHandler.Create((context) => Run(context.ParseResult));
```

Comment:
Does `Create` know that we are passing a task and this may be an async method?

Reply:
yes, it only accepts Task after recent changes, hence the change below

Reply:
Latest changes should allow both synchronous and asynchronous versions.

https://github.com/dotnet/command-line-api/blob/478fa2fe80ae912e9ca9d823738e8dbb83a01d73/src/System.CommandLine/Invocation/CommandHandler.Action.cs#L20-L34